app/src/main/java/sdp/moneyrun/LeaderboardActivity.java

```diff
@@ -53,4 +63,104 @@ public void addPlayer(Player player){
     public ArrayList<Player> getPlayerList(){
         return playerList;
     }
+
+    public void setMainPlayer(int id,String[] info){
+        int playerId = getIntent().getIntExtra("playerId",0);
+        if(id != 0)
+            playerId = id;
+        if(playerId != 0) {
+            String[] playerInfo = getIntent().getStringArrayExtra("playerId" + playerId);
+            if(info != null)
+                playerInfo = info;
+            userPlayer = new Player(playerId);
+            userPlayer.setName(playerInfo[0]);
+            userPlayer.setAddress(playerInfo[1]);
+            userPlayer.setScore(0);
+            addPlayer(userPlayer);
+        }
+    }
+    /*
+    * This function will set up players in the leaderboard once we know their player ids and names
+    * it will set up dummy players before that so that we have a leaderboard nonetheless every time a player joins
+    * the game( up to 6 players for now), on data change listeners will be attached to these players here so that
+    * once real players join the leaderboard updates accordingly
+    * */
+    public void setDummyPlayers(){
+        DatabaseProxy databaseProxy = new DatabaseProxy();
+        Player dummy1 = new Player(1000000);
+        dummy1.setName("Dummy Player 1");
+        dummy1.setAddress("Here");
+        dummy1.setScore(1);
+        //db.putPlayer(dummy1);
+        try {
+            Thread.sleep(1000);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+        addPlayer(dummy1);
+        attachListenerToPlayer(dummy1,databaseProxy);
+        for(int i = 2; i< 6;++i){
+            Player dummy = new Player(i*1000000);
+            dummy.setName("Dummy Player "+ i);
+            dummy.setAddress("Here");
+            dummy.setScore(i);
+            addPlayer(dummy);
+        }
+    }
+    private void attachListenerToPlayer(Player dummy1, DatabaseProxy databaseProxy){
+        databaseProxy.addPlayerListener(dummy1, new ValueEventListener() {
+            @Override
+            public void onDataChange(@NonNull DataSnapshot snapshot) {
+                Player update = snapshot.getValue(Player.class);
+                System.out.println(snapshot.getValue(Player.class)+ "Getting snapshot on data change in leaderboard class");
+                if(update != null)
+                    dummy1.setName(update.getName());
+            }
+
+            @Override
+            public void onCancelled(@NonNull DatabaseError error) {
+
+            }
+        });
+    }
+
+    public Player getUserPlayer() {
+        return userPlayer;
+    }
+    //    public void setUserPlayer2(){
+//        int playerId = getIntent().getIntExtra("playerId",0);
```

Comment:
I see a lot of commented code, this is also the case in the master branch, so I think it would be best not to push any commented code unless absolutely necessary because otherwise, it becomes a mess.

Reply:
Agreed on this one